Troubleshoot/TroubleshootError.ps1

```diff
@@ -383,21 +384,41 @@ if ("AKS" -eq $ClusterType ) {
     try {
         $clusterDetails = Get-AzAks -Id $ClusterResourceId -ErrorVariable clusterFetchError -ErrorAction SilentlyContinue
         Write-Host($clusterDetails | Format-List | Out-String)
-        $clusterSPNClientID = $clusterDetails.ServicePrincipalProfile.ClientId
+
+        # Check to see if SP exists, if it does use it. Else use MSI
+        if ($clusterDetails.ServicePrincipalProfile -ne $null -and $clusterDetails.ServicePrincipalProfile.ClientId -ne $null -and $clusterDetails.ServicePrincipalProfile.ClientId -ne "") {
+            Write-Host('Attempting to provide permissions to service principal...') -ForegroundColor Green
+            if ($clusterDetails.ServicePrincipalProfile.ClientId -ne "")
+            {
+                $clusterSpnClientID = $clusterDetails.ServicePrincipalProfile.ClientId
+                $isServicePrincipal = $true
+            }
+        }
+        elseif ($ResourceDetailsArray -ne $null -and $ResourceDetailsArray[0].properties.addonprofiles.omsagent -ne $null -and $ResourceDetailsArray[0].properties.addonprofiles.omsagent.identity -ne $null) {
+            Write-Host('Attempting to provide permissions to MSI...') -ForegroundColor Green
+            if ($ResourceDetailsArray[0].properties.addonprofiles.omsagent.identity.objectid -ne "")
```

Comment:
I persume, there will not be case where $ResourceDetailsArray[0].properties.addonprofiles.omsagent.identity.objectid  is null for the MSI enabled clusters.

Reply:
i want to add checks since i want to be able to anticipate incomplete values and not fail, but show a meaningful message.